gt/main.c

```diff
@@ -2209,6 +2252,70 @@ l_update_gt_lua_states(lua_State *l)
 	return 0;
 }
 
+static void
+process_return_message(lua_State *l, struct dynamic_config *dy_conf,
+	int num_succ_sent_inst)
+{
+	int i;
+	int num_gt_messages = 0;
+	size_t reply_len = 0;
+	char reply_msg[MSG_MAX_LEN];
+	int num_cmd;
+	struct dy_cmd_entry *dy_cmds[dy_conf->mailbox_burst_size];
+
+	/* Wait for all GT instances to synchronize. */
+	while (rte_atomic16_read(&dy_conf->num_returned_instances)
+			< num_succ_sent_inst)
+		rte_pause();
+
+	/* Load a set of commands from its mailbox ring. */
+	num_cmd = mb_dequeue_burst(&dy_conf->mb,
+		(void **)dy_cmds, dy_conf->mailbox_burst_size);
+
+	for (i = 0; i < num_cmd; i++) {
+		struct dy_cmd_entry *entry = dy_cmds[i];
+		switch (entry->op) {
+			case GT_UPDATE_POLICY_RETURN: {
+				size_t len;
+
+				if (dy_conf->gt == NULL) {
+					GT_LOG(ERR, "The command operation %u requires that the server runs as Grantor\n",
+						entry->op);
+					continue;
+				}
+
+				len = strlen(entry->u.gt.return_msg);
+				if (reply_len + len >= MSG_MAX_LEN)
+					GT_LOG(ERR, "The aggregated return message from GT blocks is too long\n");
+				else {
+					strncpy(reply_msg + reply_len,
+						entry->u.gt.return_msg, len);
+					reply_len += len;
+				}
+
+				num_gt_messages++;
+
+				break;
+			}
+			default:
+				GT_LOG(ERR, "Unknown command operation %u\n",
+					entry->op);
+				break;
+		}
+
+		mb_free_entry(&dy_conf->mb, dy_cmds[i]);
+	}
+
+	if (dy_conf->gt != NULL && num_gt_messages != dy_conf->gt->num_lcores) {
+		GT_LOG(WARNING,
+			"%s successfully collected only %d/%d instances\n",
+			__func__, num_gt_messages, dy_conf->gt->num_lcores);
+	}
+
+	reply_msg[reply_len] = '\0';
+	lua_pushstring(l, reply_msg);
```

Comment:
Use `lua_pushlstring()` instead. `lua_pushlstring()` doesn't require the line `reply_msg[reply_len] = '\0';` and, more importantly, eliminate the need for Lua to scan the string to find its length.